### PR TITLE
Add TextDisplay Component

### LINKED ISSUE #187: TextDisplay

### PR DESCRIPTION
**Pull request by amalyam**
## Describe your changes
This PR adds a `TextDisplay` component, which is intended for displaying a labeled piece of non-editable, informational text. It is a corollary to the `TextInput` component.

Its `labelPosition` defaults to left, and the label can also be hidden by setting `isLabelVisible` to `false`.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:
- [x] I have run `eslint` on the code
- [x] I have added JSDoc for all of my code (where applicable)
- [ ] I have added tests to cover my changes.

## Priority:
- [x] Normal
- [ ] High

## Related Issues:
Closes #187

### CODE REVIEW

File: src/Forms/TextDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  /** The id of the label tied to this text display field */
+  id: string;
+  /** Specifies the label text */
+  label: string;
+  /** The name of the text display field */
+  name: string;
+  /** The current value in the text display field */
+  value: string;
+  /** Allows you to pass in a label position property from the POSITION enum
+   * @default LEFT
```
Comment (natalynyu) [inline src/Forms/TextDisplay.tsx:21]: This is a very minor point, but can we change the references of the default label position from LEFT to POSITION.LEFT? This is just to be a little more accurate in referring to the POSITION enum.

File: src/Forms/TextDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+export interface StyledDisplayLabelProps {
+  /** Allows you to pass in a label position property from the POSITION enum
+   * @default LEFT
+   */
+  labelPosition?: POSITION;
+}
+
+export interface StyledTextDisplayProps {
```
Comment (natalynyu) [inline src/Forms/TextDisplay.tsx:37]: Is `StyledTextDisplayProps` being used anywhere? This seems like it can be safely removed.

**Review by natalynyu — CHANGES_REQUESTED** [review]
This looks great, Amalya! I just had some minor comments. Let me know if you have any questions.

File: src/Forms/TextDisplay.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  font-weight: bold;
+  margin-bottom: ${({ labelPosition }) => (labelPosition === POSITION.TOP ? '0.5em' : '0')};
+  margin-right: ${({ labelPosition }) => (labelPosition === POSITION.LEFT ? '0.5em' : '0')};
+  padding: ${fromTheme('ws', 'xsmall')};
+`;
+
+export const StyledContainer = styled.div<StyledDisplayLabelProps>`
+display: flex;
+flex-direction: ${({ labelPosition }) => (labelPosition === POSITION.LEFT ? 'row' : 'column')};
+align-items: ${({ labelPosition }) => (labelPosition === POSITION.LEFT ? 'center' : 'felx-start')};
```
[the remark this answers is not in the document]
Comment (amalyam) [reply src/Forms/TextDisplay.tsx:69]: Good catch!

File: src/Forms/TextDisplay.tsx (hunk 1 above)
Comment (amalyam) [reply src/Forms/TextDisplay.tsx:21]: Got it, wasn't sure what the right convention was here!

File: src/Forms/TextDisplay.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    && `display: flex;
+    align-items: center;
+  `}
+`;
+
+export const StyledDisplayLabel = styled.div<StyledDisplayLabelProps>`
+  font-weight: bold;
+  margin-bottom: ${({ labelPosition }) => (labelPosition === POSITION.TOP ? '0.5em' : '0')};
+  margin-right: ${({ labelPosition }) => (labelPosition === POSITION.LEFT ? '0.5em' : '0')};
+  padding: ${fromTheme('ws', 'xsmall')};
```
[the remark this answers is not in the document]
Comment (amalyam) [reply src/Forms/TextDisplay.tsx]: Good catch, definitely looks better without it!

**Comment by amalyam** [comment]
@natalynyu thanks for these corrections!

**Review by natalynyu — APPROVED** [review]
This looks great! Approved.

**Pull request merged**